File: src/Application Specific/Classes/Entity Collections/ContractsAlternative/ContractAlternativeSiteJobOfWork/ContractAlternativeSiteJobOfWork.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace ContractAlternativeSiteJobOfWorks

        Public Class ContractAlternativeSiteJobOfWork

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False
            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False
            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property
            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "ContractAlternativeSiteJobOfWork Properties"


            Private _ContractSiteJobOfWorkID As Integer = 0
            Public ReadOnly Property ContractSiteJobOfWorkID() As Integer
                Get
                    Return _ContractSiteJobOfWorkID
                End Get
            End Property
            Public WriteOnly Property SetContractSiteJobOfWorkID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ContractSiteJobOfWorkID", Value)
                End Set
            End Property


            Private _ContractSiteID As Integer = 0
            Public ReadOnly Property ContractSiteID() As Integer
                Get
                    Return _ContractSiteID
                End Get
            End Property
            Public WriteOnly Property SetContractSiteID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_ContractSiteID", Value)
                End Set
            End Property


            Private _FirstVisitDate As DateTime = Nothing
            Public Property FirstVisitDate() As DateTime
                Get
                    Return _FirstVisitDate
                End Get
                Set(ByVal Value As DateTime)
                    _FirstVisitDate = Value
                End Set
            End Property

            Private _PricePerVisit As Double = 0.0
            Public ReadOnly Property PricePerVisit() As Double
                Get
                    Return _PricePerVisit
                End Get
            End Property
            Public WriteOnly Property SetPricePerVisit() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PricePerVisit", Value)
                End Set
            End Property

            Private _AverageMileage As Double = 0.0
            Public ReadOnly Property AverageMileage() As Double
                Get
                    Return _AverageMileage
                End Get
            End Property
            Public WriteOnly Property SetAverageMileage() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_AverageMileage", Value)
                End Set
            End Property


            Private _PricePerMile As Double = 0.0
            Public ReadOnly Property PricePerMile() As Double
                Get
                    Return _PricePerMile
                End Get
            End Property
            Public WriteOnly Property SetPricePerMile() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_PricePerMile", Value)
                End Set
            End Property


            Private _IncludeMileagePrice As Boolean = False
            Public ReadOnly Property IncludeMileagePrice() As Boolean
                Get
                    Return _IncludeMileagePrice
                End Get
            End Property
            Public WriteOnly Property SetIncludeMileagePrice() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_IncludeMileagePrice", Value)
                End Set
            End Property


            Private _IncludeRates As Boolean = False
            Public ReadOnly Property IncludeRates() As Boolean
                Get
                    Return _IncludeRates
                End Get
            End Property
            Public WriteOnly Property SetIncludeRates() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_IncludeRates", Value)
                End Set
            End Property


            Private _TotalSitePrice As Double = 0
            Public ReadOnly Property TotalSitePrice() As Double
                Get
                    Return _TotalSitePrice
                End Get
            End Property
            Public WriteOnly Property SetTotalSitePrice() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_TotalSitePrice", Value)
                End Set
            End Property


#End Region

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/ContractsAlternative/ContractAlternativeSiteJobOfWork/ContractAlternativeSiteJobOfWork.cs
using System;
using System.Collections;

namespace FSM.Entity
{
    namespace ContractAlternativeSiteJobOfWorks
    {
        public class ContractAlternativeSiteJobOfWork
        {
            private DataTypeValidator _dataTypeValidator;

            public ContractAlternativeSiteJobOfWork()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            /* TODO ERROR: Skipped RegionDirectiveTrivia */
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
            /* TODO ERROR: Skipped RegionDirectiveTrivia */

            private int _ContractSiteJobOfWorkID = 0;

            public int ContractSiteJobOfWorkID
            {
                get
                {
                    return _ContractSiteJobOfWorkID;
                }
            }

            public object SetContractSiteJobOfWorkID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ContractSiteJobOfWorkID", value);
                }
            }

            private int _ContractSiteID = 0;

            public int ContractSiteID
            {
                get
                {
                    return _ContractSiteID;
                }
            }

            public object SetContractSiteID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_ContractSiteID", value);
                }
            }

            private DateTime _FirstVisitDate = default;

            public DateTime FirstVisitDate
            {
                get
                {
                    return _FirstVisitDate;
                }

                set
                {
                    _FirstVisitDate = value;
                }
            }

            private double _PricePerVisit = 0.0;

            public double PricePerVisit
            {
                get
                {
                    return _PricePerVisit;
                }
            }

            public object SetPricePerVisit
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PricePerVisit", value);
                }
            }

            private double _AverageMileage = 0.0;

            public double AverageMileage
            {
                get
                {
                    return _AverageMileage;
                }
            }

            public object SetAverageMileage
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_AverageMileage", value);
                }
            }

            private double _PricePerMile = 0.0;

            public double PricePerMile
            {
                get
                {
                    return _PricePerMile;
                }
            }

            public object SetPricePerMile
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_PricePerMile", value);
                }
            }

            private bool _IncludeMileagePrice = false;

            public bool IncludeMileagePrice
            {
                get
                {
                    return _IncludeMileagePrice;
                }
            }

            public object SetIncludeMileagePrice
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_IncludeMileagePrice", value);
                }
            }

            private bool _IncludeRates = false;

            public bool IncludeRates
            {
                get
                {
                    return _IncludeRates;
                }
            }

            public object SetIncludeRates
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_IncludeRates", value);
                }
            }

            private double _TotalSitePrice = 0;

            public double TotalSitePrice
            {
                get
                {
                    return _TotalSitePrice;
                }
            }

            public object SetTotalSitePrice
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_TotalSitePrice", value);
                }
            }


            /* TODO ERROR: Skipped EndRegionDirectiveTrivia */
        }
    }
}